### PR TITLE
Merge schedule properties with the program's runtime arguments.

### PR DESCRIPTION
**Pull request by sagarkapare**
1. Implementation of merging schedule properties with program's runtime arguments.
2. Unit test case.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java [hunk 1]
```diff
@@ -71,14 +74,21 @@ public void run(Id.Program programId, ProgramType programType, Arguments argumen
...
-    Map<String, String> userArgs;
+    Map<String, String> userArgs = Maps.newHashMap();
     Program program;
     try {
       program =  store.loadProgram(programId, ProgramType.WORKFLOW);
       Preconditions.checkNotNull(program, "Program not found");
 
-      userArgs = preferencesStore.getResolvedProperties(programId.getNamespaceId(), programId.getApplicationId(),
-                                                        programType.getCategoryName(), programId.getId());
+      String scheduleName = arguments.getOption(ProgramOptionConstants.SCHEDULE_NAME);
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java]: What is scheduleName is null?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java [hunk 2]
```diff
@@ -71,14 +74,21 @@ public void run(Id.Program programId, ProgramType programType, Arguments argumen
...
+      ScheduleSpecification spec = store.getApplication(programId.getApplication()).getSchedules().get(scheduleName);
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java]: Preconditions for spec not equal to null

File: cdap-app-fabric/src/test/java/co/cask/cdap/AppWithSchedule.java [hunk 3]
```diff
@@ -69,6 +78,11 @@ public void run() {
       LOG.info("Ran dummy action");
       try {
         TimeUnit.MILLISECONDS.sleep(500);
+
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/AppWithSchedule.java]: Remove empty line

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java [hunk 4]
```diff
@@ -628,6 +628,28 @@ public void testFlows() throws Exception {
     Assert.assertEquals(200, deleteQueues(TEST_NAMESPACE1, WORDCOUNT_APP_NAME, WORDCOUNT_FLOW_NAME));
   }
 
+  private void setAndTestRuntimeArgs(String namespace, String appId, String runnableType, String runnableId,
+                              Map<String, String> args)
+    throws Exception {
+    HttpResponse response;
+    String argString = GSON.toJson(args, new TypeToken<Map<String, String>>() {
+    }.getType());
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: Nit - The four lines above can be formatted better. How about?
```
private void setAndTestRuntimeArgs(String namespace, String appId, String runnableType, String runnableId,
                                   Map<String, String> args) throws Exception {
    HttpResponse response;
    String argString = GSON.toJson(args, new TypeToken<Map<String, String>>() {  }.getType());
```

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java [hunk 5]
```diff
@@ -628,6 +628,28 @@ public void testFlows() throws Exception {
...
+                                                         namespace);
+    response = doPut(versionedRuntimeArgsUrl, argString);
+
+    Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+    response = doGet(versionedRuntimeArgsUrl);
+
+    Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+    Map<String, String> argsRead = GSON.fromJson(EntityUtils.toString(response.getEntity()),
+                                                 new TypeToken<Map<String, String>>() {
+                                                 }.getType());
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: same here. formatting can be better.

**Comment by sreevatsanraman** [comment]
Minor comments, lgtm otherwise/

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java]: Added check in the DefaultAppConfigurer to make sure scheduleName is not null.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java (hunk 2 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java]: Added.

File: cdap-app-fabric/src/test/java/co/cask/cdap/AppWithSchedule.java (hunk 3 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/test/java/co/cask/cdap/AppWithSchedule.java]: Removed.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java (hunk 4 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: done.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java (hunk 5 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java]: done.

**Comment by sagarkapare** [comment]
Build is running here - https://builds.cask.co/browse/CDAP-DUT528-2

**Comment by sreevatsanraman** [comment]
Lgtm.

**Pull request merged**